lib/github/ldap/domain.rb

```diff
@@ -118,7 +118,14 @@ def user?(login, search_options = {})
         options = search_options.merge \
           filter: login_filter(@uid, login),
           size: 1
-        search(options).first
+
+        if @ldap.active_directory_capability?
+          # when doing a global search for a user's DN, set the search base to blank
+          options[:base] = ""
+          @ldap.global_catalog_search(options).first
+        else
+          search(options).first
+        end
```

Comment:
Agreed, this is an ugly pivot. I think it would make sense to follow the [configuration pattern](https://github.com/github/github-ldap/blob/master/lib/github/ldap.rb#L265) from `GitHub::Ldap` to set something like a `@user_search_strategy`, etc.